apps/web/src/components/happening-preview-box-server.tsx

```diff
@@ -0,0 +1,264 @@
+      ) : (
+        <div className="px-3 py-5 text-center text-lg font-medium">
+          <p>Ingen kommende {happeningTypeToString[type]}</p>
+          {/* TODO: Add funny gif */}
+        </div>
+      )}
+    </div>
+  );
+}
+
+type EventPreviewProps = {
+  event: Happening;
+};
+
+export function EventPreview({ event }: EventPreviewProps) {
+  const noDataBase = () => {
+    if (!event.registrationStart) {
+      return null;
+    }
+    if (isToday(new Date()) && isBefore(new Date(), new Date(event.registrationStart))) {
+      return "Påmelding i dag";
+    } else if (isAfter(new Date(), new Date(event.registrationStart))) {
+      return "Åpen";
+    }
+    return "Påmelding: " + format(new Date(event.registrationStart), "dd. MMM", { locale: nb });
+  };
+  const registrationStatus = async () => {
+    try {
+      const spotRanges = event.spotRanges;
+      const registrations = await db.query.registrations.findMany({
+        where: (registration) => eq(registration.happeningId, event._id),
+        with: { user: true },
+      });
+      const { maxCapacity, registeredCount } = getSpotRangeInfo(spotRanges ?? [], registrations);
+      if (!event.registrationStart) {
+        return null;
+      }
+      if (isToday(new Date()) && isBefore(new Date(), new Date(event.registrationStart))) {
+        return "Påmelding i dag";
+      } else if (isBefore(new Date(), new Date(event.registrationStart))) {
+        return "Påmelding: " + format(new Date(event.registrationStart), "dd. MMM", { locale: nb });
+      } else if (isAfter(new Date(), new Date(event.registrationStart))) {
+        return registeredCount + "/" + (maxCapacity || ("Uendelig" && "∞"));
+      }
+      return "Fullt";
+    } catch (error) {
+      return noDataBase();
+    }
+  };
+  return (
+    <Link href={`/arrangement/${event.slug}`}>
+      <div className={cn("flex h-full items-center gap-5 p-5", "hover:bg-muted")}>
+        <div className="flex w-full justify-between overflow-x-hidden">
+          <h3 className="my-auto text-sm font-semibold sm:text-xl">{event.title}</h3>
+          <ul className="text-sm md:text-base">
+            {event.date && (
+              <li className="flex w-20 justify-end">
+                <CalendarIcon className="my-auto mr-1" />
+                {format(new Date(event.date), "dd. MMM", { locale: nb })}
+              </li>
+            )}
+            <li className="flex justify-end">{registrationStatus()}</li>
+          </ul>
+        </div>
+      </div>
+    </Link>
+  );
+}
+
+type BedpresPreviewProps = {
+  bedpres: Happening;
+};
+
+export function BedpresPreview({ bedpres }: BedpresPreviewProps) {
+  const noDataBase = () => {
+    if (!bedpres.registrationStart) {
+      return null;
+    }
+    if (isToday(new Date()) && isBefore(new Date(), new Date(bedpres.registrationStart))) {
+      return "Påmelding i dag";
+    } else if (isAfter(new Date(), new Date(bedpres.registrationStart))) {
+      return "Åpen";
+    }
```

Comment:
Her Kan du bruke `isPast` og `isFuture`.

```suggestion
    if (isToday(new Date()) && isFuture(new Date(bedpres.registrationStart))) {
      return "Påmelding i dag";
    } else if (isPast(new Date(bedpres.registrationStart))) {
      return "Åpen";
    }
```